### PR TITLE
Core: Rename lib/api to lib/manager-api

### PR DESCRIPTION
**Pull request by ndelangen**
@tmeasday this is what renaming 1 package looks like, adding a package with the old name for backwards compatibility.

### CODE REVIEW

**Comment by tmeasday** [comment]
@ndelangen anything in particular I should look for in this PR or is it all just mechanical?

> should we continue to publish @storybook/api that re-exports @storybook/manager-api with a deprecation warning until 8.0, so that existing addons don't break? 

I think so. It's just too much to expect everyone to convert their addons by 7.0.

File: lib/deprecated-api/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "@storybook/api",
+  "version": "6.5.0-alpha.64",
+  "description": "Event names used in storybook core",
```
Comment (shilman) [inline lib/deprecated-api/package.json:4]:
```suggestion
  "description": "Deprecated manager API",
```

File: lib/deprecated-api/package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  "keywords": [
+    "storybook"
+  ],
+  "homepage": "https://github.com/storybookjs/storybook/tree/main/lib/core-events",
```
Comment (shilman) [inline lib/deprecated-api/package.json:8]:
```suggestion
  "homepage": "https://github.com/storybookjs/storybook/tree/main/lib/deprecated-api",
```

**Review by shilman — CHANGES_REQUESTED** [review]

File: lib/deprecated-api/package.json (hunk 1 above)
Comment (tmeasday) [reply lib/deprecated-api/package.json:4]: Maybe we should put this in `lib/deprecated/api`?

**Comment by ndelangen** [comment]
I'll try to revisit this over the weekend/Monday

**Pull request closed without merging**